mahiru/components/domain_administrator.py

```diff
@@ -180,6 +186,88 @@ def execute_step(
                 for asset in assets.values():
                     self._free_image(asset.id)
 
+    def serve_asset(
+            self, asset: Asset, connection_request: ConnectionRequest
+            ) -> ConnectionInfo:
+        """Serve an asset as a VPN-reachable service.
+
+        The asset's image_location must be a path on the local disk,
+        this function does not serve remote images.
+
+        Args:
+            asset: A local asset to serve.
+            connection_request: A description of the remote end of the
+                VPN connection to set up.
+        """
+        if asset.image_location is None:
+            raise RuntimeError(
+                    f'Cannot serve asset {asset} because it has no image')
+
+        if asset.image_location.startswith('http:'):
```

Comment:
Good point, there will be `https` at least in the future (and `http` needs to go, being insecure). Maybe it would be better to start any local paths with `file://`, test for that, and assume anything else is remote. Or maybe we should even have different Asset classes, one for communication between the sites with a URL and another one for internal use with a local path. That may be better, but doesn't belong in this PR. I've made https://github.com/SecConNet/mahiru/issues/102 for this so I can come back to it later.